### PR TITLE
Move listview settings reducer and add tests

### PR DESCRIPTION
**Pull request by HichamELBSI**
Signed-off-by: HichamELBSI <elabbassih@gmail.com>

### What does it do?

Move the list view settings reducer of the Content Manager to Immer and add tests.

### Why is it needed?

Describe the issue you are solving.

### How to test it?

Provide information about the environment and the path to verify the behaviour.

### Related issue(s)/PR(s)

Let us know if this is related to any issue/pull request

### CODE REVIEW

File: examples/getstarted/api/address/models/Address.settings.json [hunk 1]
```diff
@@ -32,7 +32,8 @@
         "videos"
       ],
       "plugin": "upload",
-      "required": false
+      "required": false,
```
Comment (mfrachet) [inline examples/getstarted/api/address/models/Address.settings.json:35]: Should we revert this file?

File: packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js [hunk 2]
```diff
@@ -1,77 +1,114 @@
-import { fromJS } from 'immutable';
+import produce, { current } from 'immer';
+import set from 'lodash/set';
+import get from 'lodash/get';
```
Comment (mfrachet) [inline packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:3]: 👍🏻 🚀

**Review by mfrachet — APPROVED** [review]
Looks concrete to me 👍🏻 GG and love the teeeeests 🚀

File: packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js [hunk 3]
```diff
@@ -1,77 +1,114 @@
...
+          layoutFieldList.filter((_, index) => action.index !== index)
+        );
 
-      return state
-        .removeIn(['modifiedData', 'layouts', 'list', action.index])
-        .updateIn(defaultSortByPath, () => {
-          if (attrToBeRemoved === defaultSortBy) {
-            return attrToSelect;
-          }
+        // TODO : Check with the team if we still need the defaultSortBy reassignment.
```
Comment (soupette) [inline packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:59]: IMO, you can remove this part

File: packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js [hunk 4]
```diff
@@ -1,77 +1,114 @@
...
-      const attrPath = ['modifiedData', 'layouts', 'list', action.index];
-      const attrToBeRemoved = state.getIn(attrPath);
+const reducer = (state = initialState, action) =>
+  // eslint-disable-next-line consistent-return
+  produce(state, draftState => {
+    const layoutFieldListPath = ['modifiedData', 'layouts', 'list'];
+    switch (action.type) {
+      case 'ADD_FIELD': {
+        const layoutFieldList = get(state, layoutFieldListPath, []);
+        set(draftState, layoutFieldListPath, [...layoutFieldList, action.item]);
```
[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:20]: `modifiedData.layouts.list` is not explicitly defined the first time so an `undefined` error will be triggered.

File: packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js [hunk 5]
```diff
@@ -1,77 +1,114 @@
...
+      case 'MOVE_FIELD': {
+        const layoutFieldList = get(state, layoutFieldListPath, []);
+        const { originalIndex, atIndex } = action;
 
-      const firstAttr = state.getIn(['modifiedData', 'layouts', 'list', 1]);
-      const firstAttrType = state.getIn(['modifiedData', 'attributes', firstAttr, 'type']);
-      const attrToSelect =
-        firstAttrType !== 'media' && firstAttrType !== 'richtext' ? firstAttr : 'id';
+        if (
+          layoutFieldList.length > 1 &&
```
[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:28]: This is just to add more security on the move field action. Also, avoid a useless set move if there is no need to.

File: packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (soupette) [reply packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:20]: Shouldn't it always be defined?

[the remark this answers is not in the document]
Comment (HichamELBSI) [reply packages/core/content-manager/admin/src/pages/ListSettingsView/reducer.js:20]: Not explicitly in the initial state. We can remove it if you think it is overkill.

**Pull request merged**